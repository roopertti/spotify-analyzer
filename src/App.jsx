import React from 'react';

import PrimaryButton from './components/common/PrimaryButton';

function initAuth() {
  console.log('Auth start');
}

export default function App() {
  return (
    <div>
      <h1>Hello world!</h1>
      <PrimaryButton onClick={initAuth}>Authorize</PrimaryButton>
    </div>
  );
}
