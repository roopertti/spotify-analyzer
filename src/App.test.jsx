import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import App from './App';

describe('App tests', () => {
  it('Render whole App', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
