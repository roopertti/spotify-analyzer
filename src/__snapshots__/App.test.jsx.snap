// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App tests Render whole App 1`] = `
Array [
  .c1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  margin: 0;
  font-family: 'Raleway',sans-serif;
  font-style: normal;
  font-weight: bold;
  font-size: 2rem;
  box-shadow: 0px 4px 30px 0px #b118c8 30%;
  background-color: #2f29bf;
  background-image: linear-gradient(90.05deg,#2f29bf -15.19%,rgba(255,255,255,0) 46.55%), linear-gradient(90.13deg,#b118c8 46.55%,rgba(255,255,255,0) 142.37%),linear-gradient(0deg,#ff8600,#ff8600);
  background-size: 100%;
  -webkit-background-clip: text;
  -moz-background-clip: text;
  -webkit-text-fill-color: transparent;
  -moz-text-fill-color: transparent;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 3rem;
  box-shadow: 0px 10px 30px rgba(47,41,191,0.3),0px -10px 30px rgba(177,24,200,0.3);
}

.c2 {
  font-size: 2rem;
  outline: none;
  z-index: 3;
  background: none;
  border-radius: 50%;
  width: 2.5rem;
  height: 2.5rem;
  color: #b118c8;
  border: none;
  -webkit-transform: none;
  -ms-transform: none;
  transform: none;
  -webkit-transition-duration: 0.3s;
  transition-duration: 0.3s;
}

.c2:hover {
  background-color: #e5e5e5;
}

.c2:focus: {
  box-shadow: linear-gradient(90.2deg,#2f29bf 0%,rgba(255,255,255,0) 100%),#b118c8;
}

@media (min-width:960px) {

}

@media (min-width:960px) {

}

@media (max-width:960px) {
  .c1 {
    font-size: 1.8rem;
  }
}

@media (max-width:600px) {
  .c1 {
    font-size: 1.5rem;
  }
}

@media (max-width:960px) {
  .c0 {
    padding: 2rem;
  }
}

@media (max-width:600px) {
  .c0 {
    padding: 1.5rem;
  }
}

@media (max-width:960px) {
  .c2 {
    font-size: 1.8rem;
  }
}

@media (max-width:600px) {
  .c2 {
    font-size: 1.5rem;
  }
}

@media (min-width:600px) {

}

@media (min-width:1200px) {

}

@media (min-width:960px) {

}

@media (min-width:960px) {

}

@media (min-width:600px) {

}

<header
    className="c0"
  >
    <h1
      className="c1"
    >
      Spotify analyzer
    </h1>
    <button
      className="c2"
      data-testid="menu-button"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-caret-down fa-w-10 "
        data-icon="caret-down"
        data-prefix="fas"
        focusable="false"
        onClick={[Function]}
        role="img"
        style={Object {}}
        viewBox="0 0 320 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </button>
  </header>,
  .c0 {
  width: 100%;
  margin: auto;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
}

.c2 {
  text-transform: uppercase;
  font-size: 0.9rem;
  font-weight: 500;
  color: white;
  border: none;
  border-radius: 50px;
  background: linear-gradient(90.2deg,#2f29bf -21.41%,rgba(255,255,255,0) 134.41%),#b118c8;
  padding: 1.2rem 4.3rem;
  box-shadow: 0px 10px 30px rgba(47,41,191,0.3),0px -10px 30px rgba(177,24,200,0.3);
  -webkit-transition-duration: 0.3s;
  transition-duration: 0.3s;
}

.c2:hover,
.c2:focus {
  box-shadow: 0px 3px 7px rgba(47,41,191,0.3),0px -3px 7px rgba(177,24,200,0.3);
}

.c2:active {
  background: linear-gradient(90.2deg,#2f29bf 0%,rgba(255,255,255,0) 100%),#b118c8;
  box-shadow: none;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-align: center;
  padding: 5rem 1rem;
}

.c1 > h1 {
  font-size: 1.5rem;
}

.c1 > h5 {
  font-size: 1rem;
  color: #343434;
  margin-bottom: 3rem;
}

@media (min-width:960px) {

}

@media (min-width:960px) {

}

@media (max-width:960px) {

}

@media (max-width:600px) {

}

@media (max-width:960px) {

}

@media (max-width:600px) {

}

@media (max-width:960px) {

}

@media (max-width:600px) {

}

@media (min-width:600px) {
  .c0 {
    width: 600px;
  }
}

@media (min-width:1200px) {
  .c0 {
    width: 960px;
  }
}

@media (min-width:960px) {
  .c2 {
    font-size: 1rem;
    padding: 1.4rem 5rem;
  }
}

@media (min-width:960px) {
  .c1 > h1 {
    font-size: 2.5rem;
  }

  .c1 > h5 {
    color: #343434;
    font-size: 1.5rem;
    margin-bottom: 5rem;
  }
}

@media (min-width:600px) {
  .c1 > h1 {
    font-size: 2rem;
  }

  .c1 > h5 {
    color: #343434;
    font-size: 1.1rem;
    margin-bottom: 4rem;
  }
}

<div
    className="c0"
  >
    <section
      className="c1"
    >
      <h1>
        Authorize with your Spotify account to analyze tracks.
      </h1>
      <h5>
        Authorization is required to access your playlists.
      </h5>
      <button
        className="c2"
        onClick={[Function]}
      >
        Authorize
      </button>
    </section>
  </div>,
  .c2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  font-size: 0.8rem;
  font-weight: 500;
  color: #2f29bf;
  border: 2px solid #2f29bf;
  border-radius: 50px;
  padding: 0.6rem 2rem;
}

.c2:hover,
.c2:focus {
  background-color: #e5e5e5;
}

.c2:active {
  border: 2px solid #b118c8;
  color: #b118c8;
}

.c3 {
  margin-right: 0.5rem;
  font-size: 1rem;
}

.c0 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0.8rem 0 1.6rem;
  box-shadow: 0px 10px 30px rgba(47,41,191,0.3),0px -10px 30px rgba(177,24,200,0.3);
}

.c1 {
  margin: 1rem 0;
}

@media (min-width:960px) {
  .c2 {
    font-size: 0.9rem;
    padding: 0.6rem 2.3rem;
  }
}

@media (min-width:960px) {
  .c3 {
    font-size: 1.1rem;
  }
}

@media (max-width:960px) {

}

@media (max-width:600px) {

}

@media (max-width:960px) {

}

@media (max-width:600px) {

}

@media (max-width:960px) {

}

@media (max-width:600px) {

}

@media (min-width:600px) {

}

@media (min-width:1200px) {

}

@media (min-width:960px) {

}

@media (min-width:960px) {

}

@media (min-width:600px) {

}

<footer
    className="c0"
  >
    <p
      className="c1"
    >
      Created by Robert Kuhlmann
    </p>
    <a
      className="c2"
      href="#"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-github fa-w-16 c3"
        data-icon="github"
        data-prefix="fab"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 496 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
      <span>
        Source code
      </span>
    </a>
  </footer>,
]
`;
